### PR TITLE
docs: enhance watcher k3d setup

### PR DESCRIPTION
**Pull request by adityabhatia**

### CODE REVIEW

File: docs/developer/watcher-local-test.md [hunk 1]
```diff
@@ -1,142 +1,167 @@
...
-make local-deploy-with-watcher IMG=eu.gcr.io/kyma-project/lifecycle-manager:latest
-```
+4. Create `module-template` by using [kyma-cli](https://github.com/kyma-project/cli)
+   which serves the role of a component-descriptor for module installations.
+    
+   For this setup, we will create a module template from the [template-operator](https://github.com/kyma-project/template-operator) repository as reference.
+   Adjust your path to your template-operator local directory or any other reference module operator accordingly.
+
+    ```shell
+   kyma alpha create module -p ../template-operator --version 1.2.3 -w --registry k3d-registry.localhost:5111 --insecure
```
Comment (khlifi411) [inline docs/developer/watcher-local-test.md:32]: I am getting this error in this step:
```log
\ Validating Default CRError: Error applying the default CR: failed to create typed patch object (default/sample-sample; operator.kyma-project.io/v1alpha1, Kind=Sample): .spec.releaseName: field not declared in schema
```

File: docs/developer/watcher-local-test.md [hunk 2]
```diff
@@ -1,142 +1,167 @@
...
-```
+### KCP cluster setup
+   
+1. Create a local control-plane (KCP) cluster:
+    ```shell
+    k3d cluster create kcp-local --port 9080:80@loadbalancer \
+    --registry-create k3d-registry.localhost:0.0.0.0:5111 \
+    --k3s-arg '--no-deploy=traefik@server:0'
+    ```
+2. Deploy lifecycle-manager on the cluster:
```
Comment (khlifi411) [inline docs/developer/watcher-local-test.md:11]: This command complains about missing module-manager CRDs. Could you place it after the step where we are installing module-manager CRDs.

**Review by khlifi411 — CHANGES_REQUESTED** [review]
couldn't test it locally yet

File: docs/developer/watcher-local-test.md [hunk 3]
```diff
@@ -1,142 +1,167 @@
...
 
-### Apply sample module templates for sample-kyma:
-Run the following commands to apply sample module template needed for sample Kyma on the KCP cluster:
-```shell
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/lifecycle-manager/main/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_kcp-module.yaml
-```
+    ```shell
+   kyma alpha create module -p ../template-operator --version 1.2.3 -w --registry k3d-registry.localhost:5111 --insecure
+   ```
+4. The previous step will create a `template.yaml` file in the root directory, which is the `module-template`, apply it to the cluster
```
Comment (khlifi411) [inline docs/developer/watcher-local-test.md:30]: This step is complaining about moduletemplates CRDs after we moved the lifecycle manager deploy to step 6. I would suggest to move steps 3 and 4 after that.

File: docs/developer/watcher-local-test.md [hunk 4]
```diff
@@ -1,142 +1,167 @@
...
+      channel: regular
+      sync:
+        enabled: true
+      modules:
+      - name: template-operator
+    EOF
+    ```
+
+### Watcher installation verification
+By checking the `Kyma CR` events, verify that the `SKRWebhookIsReady` ready condition is set to `True`
```
Comment (khlifi411) [inline docs/developer/watcher-local-test.md:125]: Kyma is stuck in `Error` state with the following event logs:
```log
Warning  ModuleReconciliationError  3s (x6 over 6s)  lifecycle-manager  could not synchronize remote module catalog: could not synchronize remote module catalog: could not apply catalog templates: could not apply module template diff: namespaces "kcp-system" not found
```

Comment (adityabhatia) [reply docs/developer/watcher-local-test.md:125]: for this the control-plane property should be set on the module-template

File: docs/developer/watcher-local-test.md [hunk 5]
```diff
@@ -1,142 +1,167 @@
...
-kubectl apply -f https://raw.githubusercontent.com/kyma-project/lifecycle-manager/main/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_kcp-module.yaml
-```
+5. Create `module-template` by using [kyma-cli](https://github.com/kyma-project/cli)
+   which serves the role of a component-descriptor for module installations.
+
+   For this setup, we will create a module template from the [template-operator](https://github.com/kyma-project/template-operator) repository as reference.
+   Adjust your path to your template-operator local directory or any other reference module operator accordingly.
+
+    ```shell
+   kyma alpha create module -p ../template-operator --version 1.2.3 -w --registry k3d-registry.localhost:5111 --insecure
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/developer/watcher-local-test.md:44]: The latest kyma-cli issues the following prompt here:
```
WARNING: This command is experimental and might change in its final version. Use at your own risk.
- Kustomize ready
? Please specify the file to use as default CR in /[edited]/golang/src/github.com/kyma-project/template-operator/config/samples: [operator.kyma-project.io_v1alpha1_sample.yaml operator.kyma-project.io_v1alpha1_samplehelm.yaml]
```
Which CR should be used?

Comment (adityabhatia) [reply docs/developer/watcher-local-test.md:44]: You can use any CR. I think the choice comes with the latest merge in https://github.com/kyma-project/template-operator/pull/14

File: docs/developer/watcher-local-test.md [hunk 6]
```diff
@@ -1,142 +1,172 @@
...
+      ca.crt: $(cat ca.pem | base64)
+      tls.crt: $(cat signed-cert.pem | base64)
+      tls.key: $(cat signed-cert-key.pem | base64)
+    EOF
+    ```
 
-### Create sample Kyma CR
-Run the following command to generate and apply sample Kyma CR and its corresponding secret on KCP:
+### SKR cluster setup
+Create a local kyma-runtime (SKR) cluster and a `kyma-system` namespace inside.
```
Comment (khlifi411) [inline docs/developer/watcher-local-test.md:84]:
```suggestion
Create a local kyma-runtime (SKR) cluster and a `kcp-system` namespace inside.
```

File: docs/developer/watcher-local-test.md [hunk 7]
```diff
@@ -1,142 +1,172 @@
...
+
+### Watcher event trigger
+1. Switch the context for using SKR cluster
+    ```shell
+    kubectl config use-context k3d-skr-local
+    ```
+2. Change the channel of the `template-operator` module to trigger a watcher event to KCP
+    ```yaml
+      modules:
+      - name: kcp-module
```
Comment (khlifi411) [inline docs/developer/watcher-local-test.md:146]:
```suggestion
      - name: template-operator
```

**Review by khlifi411 — APPROVED** [review]

**Pull request merged**